src/test/java/org/gridsuite/sensitivityanalysis/server/service/SensitivityAnalysisServiceTest.java

```diff
@@ -0,0 +1,368 @@
+            // l1 x GEN, before + a1, a2, a3 (implicit)
+            new SensitivityValue(0, -1, 500.1, 2.9),
+            new SensitivityValue(0, 0, 500.3, 2.7),
+            new SensitivityValue(0, 1, 500.4, 2.6),
+            new SensitivityValue(0, 2, 500.5, 2.5),
+            // l2 x GEN, just before
+            new SensitivityValue(1, -1, 500.2, 2.8),
+            // l3 x LOAD, before + a1, a2, a3 (explicit)
+            new SensitivityValue(2, -1, 500.9, 2.1),
+            new SensitivityValue(specific ? 3 : 2, 1, 500.6, 2.4),
+            new SensitivityValue(specific ? 4 : 2, 0, 500.7, 2.3),
+            new SensitivityValue(specific ? 5 : 2, 2, 500.8, 2.2)
+        ));
+        Collections.shuffle(sensitivityValues);
+
+        final SensitivityAnalysisResult result = new SensitivityAnalysisResult(sensitivityFactors,
+            contingenciesStatuses,
+            sensitivityValues);
+
+        final UUID resultUuid = UUID.randomUUID();
+        given(uuidGeneratorService.generate()).willReturn(resultUuid);
+        given(RUNNER.runAsync(eq(NETWORK), eq(VARIANT_ID), anyList(), anyList(), anyList(),
+            any(SensitivityAnalysisParameters.class), any(ComputationManager.class), any(Reporter.class)))
+            .willReturn(CompletableFuture.completedFuture(result));
+
+        UUID gottenResultUuid = analysisService.runAndSaveResult(
+            new SensitivityAnalysisRunContext(NETWORK_UUID, VARIANT_ID,
+                Collections.emptyList(), getDummyInputData(), null, null, null, null));
+        assertThat(gottenResultUuid, not(nullValue()));
+        assertThat(gottenResultUuid, is(resultUuid));
+
+        ResultsSelector.ResultsSelectorBuilder builder = ResultsSelector.builder();
+        ResultsSelector selectorN = builder
+            .isJustBefore(true)
+            .functionType(MW_FUNC_TYPE)
+            .functionIds(Set.of("l1", "l2", "l3"))
+            .variableIds(Set.of("GEN", "LOAD"))
+            .sortKeysWithWeightAndDirection(Map.of(
+                ResultsSelector.SortKey.SENSITIVITY, -1,
+                ResultsSelector.SortKey.REFERENCE, 2,
+                ResultsSelector.SortKey.VARIABLE, 3,
+                ResultsSelector.SortKey.FUNCTION, 4))
+            .build();
+
+        SensitivityRunQueryResult gottenResult;
+        gottenResult = analysisService.getRunResult(resultUuid, selectorN);
+        assertThat(gottenResult, not(nullValue()));
+        List<? extends SensitivityOfTo> sensitivities = gottenResult.getSensitivities();
+        assertThat(sensitivities, not(nullValue()));
+        assertThat(sensitivities.size(), is(3));
+        assertThat(sensitivities, Every.everyItem(Matchers.any(SensitivityOfTo.class)));
+        List<Double> sensitivityVals = sensitivities.stream().map(SensitivityOfTo::getValue).collect(Collectors.toList());
+        assertThat(sensitivityVals, isOrderedAccordingTo(Comparator.<Double>reverseOrder()));
```

Comment:
why creating OrderMatcher class for that ?

Guava Comparators class can do that : 

boolean sorted = Comparators.isInOrder(list, comparator);

Reply:
Did not know a lot about Guava, since Convergence does not use it. And did not know it was allowed.

Reply:
IntelliJ tells me
'isInOrder(java.lang.Iterable<? extends T>, java.util.Comparator<T>)' is marked unstable with @Beta 
and its interface does not play well with assertThat, as the collection is to be given as first argument and comparator as second argument, where hamcrest matchers are expected (interface) to have a `matches(Object)` method and `public void describeMismatch(Object item, Description description)`

Reply:
assertTrue(Comparators.isInOrder(sensitivityVals, Comparator.naturalOrder()));    ?

Reply:
is the beta annotation really a pb ?

Reply:
```
assertThat(List.of(2.0, 1.0), isOrderedAccordingTo(Comparator.<Double>naturalOrder()));
```
(which is not true, that's just to show the diagnostic), yields : 
```
java.lang.AssertionError: 
Expected: Checks order of an iterable according to a given comparator
     but: <2.0> and <1.0> are not ordered as given comparator expects
```
which is informative, when
```
assertTrue(Comparators.isInOrder(List.of(2.0, 1.0), Comparator.naturalOrder()));
```
yields
```
java.lang.AssertionError
```
with just the stack trace, which is slighter less informative, no ?